ddl/ddl_api.go

```diff
@@ -440,6 +441,10 @@ func columnDefToCol(ctx sessionctx.Context, offset int, colDef *ast.ColumnDef, o
 				col.GeneratedStored = v.Stored
 				_, dependColNames := findDependedColumnNames(colDef)
 				col.Dependences = dependColNames
+			case ast.ColumnOptionCollate:
+				if field_types.HasCharset(colDef.Tp) {
+					col.FieldType.Collate = v.StrValue
```

Comment:
Just set collate, keep the charset empty?

Reply:
well, I think maybe another PR is needed to solve the correlation between charset and collation.